src/main/java/org/elasticsearch/index/query/TermsFilterParser.java

```diff
@@ -181,15 +181,12 @@ public Filter parse(QueryParseContext parseContext) throws IOException, QueryPar
         }
 
         if (lookupId != null) {
-            // if there are no mappings, then nothing has been indexing yet against this shard, so we can return
-            // no match (but not cached!), since the Terms Lookup relies on the fact that there are mappings...
-            if (fieldMapper == null) {
-                return Queries.MATCH_NO_FILTER;
+            final TermsLookup lookup = new TermsLookup(lookupIndex, lookupType, lookupId, lookupRouting, lookupPath, parseContext);
+            final GetResponse getResponse = client.get(new GetRequest(lookup.getIndex(), lookup.getType(), lookup.getId()).preference("_local").routing(lookup.getRouting())).actionGet();
+            if (getResponse.isExists()) {
+                List<Object> values = XContentMapValues.extractRawValues(lookup.getPath(), getResponse.getSourceAsMap());
+                terms.addAll(values);
             }
```

Comment:
> what do we do if isExists() returns false?

If there is no such entry in the index, then we will call the terms filter parser on an empty list, and this generates a MatchNoDocsFilter (see AbstractFieldMapper.termsFilter).

> I know it might be a preexisting issue but if we execute this on a network thread we might deadlock?

I did not change the way that the request is performed, so if there is an issue, it is pre-existing indeed. Any suggestion on how to fix the issue (I don't know much about how we deal with thread pools and network threads)? If it's not easy to fix, I'd lean towards opening another issue for it if you don't mind?